src/platform_impl/android/mod.rs

```diff
@@ -299,6 +302,17 @@ impl<T: 'static> EventLoop<T> {
             match self.first_event.take() {
                 Some(EventSource::Callback) => match ndk_glue::poll_events().unwrap() {
                     Event::WindowCreated => {
+                        // Acquire a lock on the window to prevent Android from destroying
+                        // it until we've notified and waited for the user in Event::Suspended.
+                        // XXX: Don't send this event if Android already removed the window
+                        // before we managed to handle this event?
```

Comment:
Right; ndk-glue's design is still fundamentally unsynchronized so it looks like there is still a race condition here and you could potentially get a panic through no fault of the application.

I could vaguely imagine a situation where you switch to an app and then hit something like 'home' which could immediately suspend the app and if the device was pretty busy with destroying another app and doesn't really have time to think it seems plausible an app could get a window create + terminated delivered before the OS ends up scheduling the rust thread if you're unlucky.

This is still going to be better than the current situation I guess, so maybe still worth landing in the short term.

Maybe this comment should instead highlight that there's still a race condition here?

Maybe it should be possible to "try" and get the native window instead, so the race could be handled more gracefully without a panic?

Reply:
> Maybe this comment should instead

Hence `XXX`.